github.go

```diff
@@ -198,8 +199,66 @@ func (m *GithubClient) PostComment(prNumber, comment string) error {
 	return err
 }
 
+// GetChangedFiles ...
+func (m *GithubClient) GetChangedFiles(prNumber string, commitRef string) ([]ChangedFileObject, error) {
+	pr, err := strconv.Atoi(prNumber)
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert pull request number to int: %s", err)
+	}
+
+	var cfo []ChangedFileObject
+
+	var filequery struct {
+		Repository struct {
+			PullRequest struct {
+				Files struct {
+					Edges []struct {
+						Node struct {
+							ChangedFileObject
+						}
+					} `graphql:"edges"`
+					PageInfo struct {
+						PageInfoObject
+					} `graphql:"pageInfo"`
+				} `graphql:"files(first:$changedFilesFirst, after: $changedFilesEndCursor)"`
+			} `graphql:"pullRequest(number:$prNumber)"`
+		} `graphql:"repository(owner:$repositoryOwner,name:$repositoryName)"`
+	}
+
+	offset := ""
+	fetch := true
+
+	for fetch {
+		vars := map[string]interface{}{
+			"repositoryOwner":       githubv4.String(m.Owner),
+			"repositoryName":        githubv4.String(m.Repository),
+			"prNumber":              githubv4.Int(pr),
+			"changedFilesFirst":     githubv4.Int(100),
+			"changedFilesEndCursor": githubv4.String(offset),
+		}
+
+		if err := m.V4.Query(context.TODO(), &filequery, vars); err != nil {
+			return nil, err
+		}
+
+		for _, f := range filequery.Repository.PullRequest.Files.Edges {
+			cfo = append(cfo, ChangedFileObject{Path: f.Node.Path})
+		}
+
+		if filequery.Repository.PullRequest.Files.PageInfo.HasNextPage {
+			fetch = true
+			offset = filequery.Repository.PullRequest.Files.PageInfo.EndCursor
+		} else {
+			fetch = false
+		}
+
+	}
+
+	return cfo, nil
+}
+
 // GetPullRequest ...
-func (m *GithubClient) GetPullRequest(prNumber, commitRef string) (*PullRequest, error) {
+func (m *GithubClient) GetPullRequest(prNumber, commitRef string, changedFileQuery bool) (*PullRequest, error) {
```

Comment:
I think it makes sense to implement this in `in.go` instead of adding the condition to this function signature, since it ends up being two API calls anyway.

Reply:
Sure